mlibrary/search
# Resolving unnecessary rerenders.

# Overview
While developing Search, this warning popped up multiple times in console:
```
Selector unknown returned a different result when called with the same parameters. This can lead to unnecessary rerenders.
Selectors that return a new reference (such as an object or an array) should be memoized: https://redux.js.org/usage/deriving-data-selectors#optimizing-selectors-with-memoization
```

This pull request resolves all the warnings and has simplified many affected components.

## Anything else?
### `regenerator-runtime`
`regenerator-runtime` has been uninstalled due to not being used.

### `stringifySearch`
`stringifySearch` has been created to control all use cases of repeated `qs.stringify...`.

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- Run the site and poke around to see if everything still works.
